Client Profile Docs upload (DEV-89)

Davit-BetterAngels opened the pull request:
DEV-664 replaced with DEV-89

mikefeldberg commented on libs/expo/shared/ui-components/src/lib/LibraryModal/Camera.tsx:
```diff
@@ -0,0 +1,187 @@
+import { ReactNativeFile } from '@monorepo/expo/shared/apollo';
```
why do we need this and `libs/expo/shared/ui-components/src/lib/CameraPicker/index.tsx`?

Davit-BetterAngels replied:
updated camera for interaction and doc uploads. 
Now we have Camera component. but as each has different modals the modals will be separate. 
please take a look

mikefeldberg commented on libs/expo/shared/ui-components/src/lib/LibraryModal/index.tsx:
```diff
@@ -0,0 +1,166 @@
+                height: 2,
+              },
+              shadowOpacity: 0.25,
+              shadowRadius: 3.84,
```
this is a very specific number. Can we use one of our existing constants here? We have an `xxs: 4`. I see this `3.84` in a few other places already in the code and I wonder if we can replace all of those, too

Davit-BetterAngels replied:
3.84 is Material design specific elevation. It is very common usage. 
only that is why

mikefeldberg replied:
oh, I see. If it's that common, can we abstract it to a constant?

Davit-BetterAngels replied:
what if I create a constant for a shadow? like an object. so we will just spread that everywhere shadow is needed

Davit-BetterAngels replied:
created a constant object Shadow

Davit-BetterAngels commented on libs/expo/betterangels/src/lib/screens/Client/Docs/UploadModal/DriverLicense/index.tsx:
```diff
@@ -0,0 +1,363 @@
+  });
+
+  const uploadDocument = async () => {
+    if (!docs?.driverLicenseFront || !docs.driverLicenseBack || !client) {
```
he can't. that was a requirement

Davit-BetterAngels commented on libs/expo/betterangels/src/lib/screens/Client/Docs/UploadModal/BirthCertificate/index.tsx:
```diff
@@ -0,0 +1,206 @@
+      return;
+    }
+
+    const fileToUploadFront = new ReactNativeFile({
```
yep. copy past issue

Davit-BetterAngels commented on libs/expo/shared/ui-components/src/lib/BottomActions/index.tsx:
```diff
@@ -8,13 +8,20 @@ interface IBottomActionsProps {
   cancel: ReactNode;
```
if not how can we cancel the upload ?

mikefeldberg replied:
This isn't adding cancel functionality though, it's adding disable on load. Which is important, but it's not part of this ticket

mikefeldberg replied:
This isn't adding cancel functionality though, it's adding disable on load. Which is important, but it's not part of this ticket

Davit-BetterAngels commented on libs/expo/betterangels/src/lib/screens/Client/Docs/UploadModal/types.ts:
```diff
@@ -0,0 +1,42 @@
+  incomeForms: ReactNativeFile[] | undefined;
+};
+
+export type ITab =
```
dl has 2, even if I write driverLicense it will be different because it contains 2 front and back. 
bc was too long I wrote shorten as ssn.

Davit-BetterAngels replied:
updated

mikefeldberg replied:
> dl has 2, even if I write driverLicense it will be different because it contains 2 front and back. bc was too long I wrote shorten as ssn.

should this list have both `DriversLicenseFront` and `DriversLicenseBack`, in that case?

mikefeldberg replied:
> dl has 2, even if I write driverLicense it will be different because it contains 2 front and back. bc was too long I wrote shorten as ssn.

should this list have both `DriversLicenseFront` and `DriversLicenseBack`, in that case?

Davit-BetterAngels replied:
no because that is a tab and not a file

Davit-BetterAngels commented on libs/expo/shared/ui-components/src/lib/Button/Button.tsx:
```diff
@@ -154,6 +154,8 @@ interface IButtonProps {
   borderRadius?: 8 | 50;
```
Yes. UI requires another border radius

Davit-BetterAngels commented on libs/expo/betterangels/src/lib/screens/Client/Docs/UploadModal/BirthCertificate/index.tsx:
```diff
@@ -0,0 +1,206 @@
+import { ReactNativeFile } from '@monorepo/expo/shared/apollo';
```
updated with reusable

mikefeldberg commented on libs/expo/betterangels/src/lib/screens/Client/Docs/UploadModal/index.tsx:
```diff
@@ -0,0 +1,474 @@
+                </View>
+
+                <Button
+                  disabled={
```
I know this is consistent with the [design](https://www.figma.com/design/waIeX8dknHKedTrVxCQAAG/HIFI---1st-MVPs?node-id=14541-93420&t=4qSbzIWUBRl80Xjw-4), but I think we should change it for the MVP. My reasoning is: this makes it impossible to upload a new version of a document before deleting the existing version. But the MVP doesn't include "delete" functionality, so this would make it impossible to correct a mistake.

mikefeldberg commented on libs/expo/shared/ui-components/src/lib/LibraryModal/index.tsx:
```diff
@@ -0,0 +1,166 @@
+import { useState } from 'react';
+import { Alert, Pressable, View } from 'react-native';
+import Modal from 'react-native-modal';
+import LibCamera from '../Camera';
```
Is there a reason this needs to be aliased? Can't we just call it `Camera` the way we do in `libs/expo/shared/ui-components/src/lib/InteractionCamera/index.tsx`? If it _does_ have to be aliased, do we want to alias it in both places? e.g., `DocLibraryCamera` and `InteractionCamera`

mikefeldberg commented on libs/expo/shared/ui-components/src/lib/LibraryModal/index.tsx:
```diff
@@ -0,0 +1,166 @@
+import { useState } from 'react';
+import { Alert, Pressable, View } from 'react-native';
+import Modal from 'react-native-modal';
+import LibCamera from '../Camera';
```
Is there a reason this needs to be aliased? Can't we just call it Camera the way we do in libs/expo/shared/ui-components/src/lib/InteractionCamera/index.tsx? If it does have to be aliased, do we want to alias it in both places? e.g., DocLibraryCamera and InteractionCamera

mikefeldberg replied:
similar comment.. can we rename `LibraryModal` to something more specific, like `ClientDocumentModal`. I feel like "library" is a very general word and can be confusing

Davit-BetterAngels replied:
it is general component. not specific for client. 
we can use it everywhere. maybe MediaPickerModal ?

mikefeldberg replied:
`MediaPickerModal` sgtm

Davit-BetterAngels replied:
ok

mikefeldberg commented on libs/expo/betterangels/src/lib/screens/Client/Docs/UploadModal/index.tsx:
```diff
@@ -0,0 +1,474 @@
+                </View>
+
+                <Button
+                  disabled={
```
I know this logic is consistent with the [design](https://www.figma.com/design/waIeX8dknHKedTrVxCQAAG/HIFI---1st-MVPs?node-id=14541-93420&t=4qSbzIWUBRl80Xjw-4), but I think we should change it for the MVP. My reasoning is: this makes it impossible to upload a new version of a document before deleting the existing version. But the MVP doesn't include "delete" functionality, so this would make it impossible to correct a mistake.

mikefeldberg replied:
don't make this change yet, actually. Let's check with prod. Because we were trying to scope down the MVP for the SELAH deadline, but there is no deadline anymore

mikefeldberg commented on libs/expo/betterangels/src/lib/screens/Client/Docs/UploadModal/index.tsx:
```diff
@@ -0,0 +1,474 @@
+    PhotoId: (
+      <IdDocUploads docType="PhotoId" title="Upload Photo ID" {...docProps} />
+    ),
+    SocialSecurityCard: (
```
why do `SocialSecurityCard` and `PhotoId` use `IdDocUploads` instead of `SingleDocUploads`?

Davit-BetterAngels replied:
in the future I'm sure we will have another view of taking pictures for Ids.

mikefeldberg replied:
I don't think I understand. We have a screen where the user needs to upload one object (`SingleDocUploads`), and we have a screen where the user needs to upload multiple objects (`MultipleDocUploads`). Why do we need `IdDocUploads`?

mikefeldberg replied:
@Davit-BetterAngels pasting [this comment](https://github.com/BetterAngelsLA/monorepo/pull/519#issuecomment-2346924967) here so it's in the thread
> as IDS have different size and later we will implement when IDS camera will have outlined view to take a picture and maybe picking will have a crop we have separate component for IDS rather than singleUploads and multiple uploads

mikefeldberg replied:
> IDS have different size
we can pass the size as a param, though. The shared component can have a hash like.

```
THUMBNAIL_SIZES {
    "large": large_dimensions,
    "medium": medium_dimensions,
    "small": small_dimensions,
}
```
and the shared component can take a `thumbnailSize` param

> maybe picking will have a crop
maybe, but we shouldn't be duplicating components because we might do something in the future. If we need a separate component in the future, we can create it then

Davit-BetterAngels replied:
done

mikefeldberg commented on libs/expo/betterangels/src/lib/screens/Client/Docs/UploadModal/Section.tsx:
```diff
@@ -0,0 +1,57 @@
+import { ChevronLeftIcon } from '@monorepo/expo/shared/icons';
```
Can we change `Section` to something more descriptive? e.g. `UploadScreen`

Davit-BetterAngels replied:
changed to UploadSection

mikefeldberg reviewed (commented):
thanks for making the changes. A few more comments but looks good overall 👍

mikefeldberg commented on libs/expo/betterangels/src/lib/screens/Client/Docs/UploadModal/index.tsx:
```diff
@@ -0,0 +1,474 @@
+                    justifyContent: 'center',
+                  }}
+                >
+                  {!!docs.ConsentForm && docs.ConsentForm.length > 0 && (
```
now that I think about it, this doesn't actually make sense. I can see wanting to do this for unique docs, like license and birth cert, but it doesn't make sense for forms, that aren't unique. see: [slack thread](https://betterangels-group.slack.com/archives/C05ECN9FLGL/p1725915484096409)

mikefeldberg replied:
@Davit-BetterAngels let's remove `disabled` for most of these buttons. The only buttons with `disabled` logic should be: license, birth cert, photoid, and social. Everything else should always be enabled.

Davit-BetterAngels replied:
done

Davit-BetterAngels commented:
as IDS have different size and later we will implement when IDS camera will have outlined view to take a picture and maybe picking will have a crop we have separate component for IDS rather than singleUploads and multiple uploads

shiozakiyuko commented:
Hi @mikefeldberg!

Here are my feedback with screenshots.
1. After upload a file and come back to the same section e.g. HMIS Form, you don't see the file. It's empty with X button.
2. When I tap the upload button to bring up the From Photo Album and Take Photo (and Cancel) buttons, the low opacity background is missing. (the [mockup:Figma link](https://www.figma.com/design/waIeX8dknHKedTrVxCQAAG/HIFI---1st-MVPs?node-id=15450-301639&t=yeXTJaAmzOBNMF0o-4))
3. The color of border is different from the mockup (the color should be #D3D9E3 or Neutral90)
4. When I open the Upload Files drawer, I saw some of the files are **already** "uploaded" even though I haven't uploaded those files. e.g. Driver's License (front and back), Photo ID, etc.
5. The clear button (x) inside of the text field is missing
6. The file names are not correctly shown. (Sample User: Kilgore Trout > Consent Forms)
Even though I renamed the file name, some extra name comes with it before the custom file name ("Otter"). e.g. attachments/5b7c6093-950c-47c4-bc79-84827065062a.**Otter** or attachments/c1909d07-16bc-428e-837c-307aa6277c4f.**Consent_form_01** 
Once I upload a file, it seems like the action duplicates extra files. I have uploaded a few files in Consent Forms but now I see about 20+ files. 

![image](https://github.com/user-attachments/assets/051f51c9-fbf0-4c8a-8d88-6b74e7af5acc)
![image](https://github.com/user-attachments/assets/9b3f055d-1f86-4d71-b56c-15efdf3800a0)
![image](https://github.com/user-attachments/assets/1cd09314-c5d7-4b16-ad76-b0b3782accbe)
![image](https://github.com/user-attachments/assets/5016e94a-8f86-4665-920e-c559dd857a09)
![image](https://github.com/user-attachments/assets/d4f2f28f-2ef5-48b2-bff2-946fc807c66d)
![image](https://github.com/user-attachments/assets/90b60df9-43e9-4522-8f54-da8a990a9cfc)


I wrote a bunch but if you have any questions or you want me walk through this, please let me know. Thanks!

mikefeldberg commented:
> Hi @mikefeldberg!
> 
> Here are my feedback with screenshots.
> 
> 1. After upload a file and come back to the same section e.g. HMIS Form, you don't see the file. It's empty with X button.
> 2. When I tap the upload button to bring up the From Photo Album and Take Photo (and Cancel) buttons, the low opacity background is missing. (the [mockup:Figma link](https://www.figma.com/design/waIeX8dknHKedTrVxCQAAG/HIFI---1st-MVPs?node-id=15450-301639&t=yeXTJaAmzOBNMF0o-4))
> 3. The color of border is different from the mockup (the color should be #D3D9E3 or Neutral90)
> 4. When I open the Upload Files drawer, I saw some of the files are **already** "uploaded" even though I haven't uploaded those files. e.g. Driver's License (front and back), Photo ID, etc.
> 5. The clear button (x) inside of the text field is missing
> 6. The file names are not correctly shown. (Sample User: Kilgore Trout > Consent Forms)
>    Even though I renamed the file name, some extra name comes with it before the custom file name ("Otter"). e.g. attachments/5b7c6093-950c-47c4-bc79-84827065062a.**Otter** or attachments/c1909d07-16bc-428e-837c-307aa6277c4f.**Consent_form_01**
>    Once I upload a file, it seems like the action duplicates extra files. I have uploaded a few files in Consent Forms but now I see about 20+ files.
> 
> screenshots
> I wrote a bunch but if you have any questions or you want me walk through this, please let me know. Thanks!

thanks @shiozakiyuko! For 4, I think you're seeing that because someone already uploaded those documents for the user looking at. So buttons are disabled. @Davit-BetterAngels can you please look into Yuko's remaining comments?

Davit-BetterAngels commented:
> 2\. mockup:Figma link

1. screen was not ready for edit mode. we had only create mode. work needs to be done to support edit
2. needs to be clarified. Our secondary button pressed state has grey background. we want pressed grey background and after keep blue background as selected one? or we want to remove pressed state for "secondary" button and keep only selected ?
3. fixed
4. someone uploaded before you
5. fixed
6. same answer as (1)

shiozakiyuko commented:
Hey @Davit-BetterAngels 
As for #1, I had the mockup in the figma file already. Wasn't it enough? 
<img width="252" alt="image" src="https://github.com/user-attachments/assets/dee60d77-86b3-4b58-8d7a-2def79907051">

mikefeldberg commented:
> Hey @Davit-BetterAngels As for #1, I had the mockup in the figma file already. Wasn't it enough? <img alt="image" width="252" src="https://private-user-images.githubusercontent.com/126514913/367841727-dee60d77-86b3-4b58-8d7a-2def79907051.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY1MDYwODgsIm5iZiI6MTcyNjUwNTc4OCwicGF0aCI6Ii8xMjY1MTQ5MTMvMzY3ODQxNzI3LWRlZTYwZDc3LTg2YjMtNGI1OC04ZDdhLTJkZWY3OTkwNzA1MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxNlQxNjU2MjhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04NGMyOGFmMmNiNTQ1MzU3ZWQxYzdjNmI5M2MyNWQ3NzUzNjRmMWEwMGNjY2VjZjRlZTA4ZDM0M2VlNThkODhlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.4ZGN9ef6FYqy7OqYaXM-9xK7EQAcK5H5C4IBmegkN7s">

@Davit-BetterAngels I don't think Yuko's saying that she can't view the uploads in the doc library, I think she's saying she doesn't see the thumbnail of the uploaded doc in the upload section

Davit-BetterAngels commented:
> Hey @Davit-BetterAngels As for #1, I had the mockup in the figma file already. Wasn't it enough? <img alt="image" width="252" src="https://private-user-images.githubusercontent.com/126514913/367841727-dee60d77-86b3-4b58-8d7a-2def79907051.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY1MDYwOTUsIm5iZiI6MTcyNjUwNTc5NSwicGF0aCI6Ii8xMjY1MTQ5MTMvMzY3ODQxNzI3LWRlZTYwZDc3LTg2YjMtNGI1OC04ZDdhLTJkZWY3OTkwNzA1MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxNlQxNjU2MzVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iNmRkNmUyOGYxNjJlNzAyMTQ1YzEwOTVmYzczYjQ4NTQyYzZkMzYwYTliMDRiNzg3MTI4ZjNmZWE1OGE2MTZlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.A-gua6ZQ8K1yPuwBY-YgrfqqKyLxm6YcuU3rQGotmkM">

this PR is only for Uploading and not editing. That what ticket says. Its not about figma. We can create a ticket for edit uploaded files.
